### PR TITLE
feat(kafka.js): add producer connect support

### PR DESCRIPTION
**Pull request by idonava**

### CODE REVIEW

**Review by ranrib — DISMISSED** [review]
👏

File: src/events/kafkajs.js [hunk 1]
```diff
@@ -93,6 +93,63 @@ function kafkaMiddleware(messages, producer) {
...
+function kafkaConnectMiddleware(producer) {
+    let originalHandlerAsyncError;
+    let kafkaEvent;
+    let startTime;
+    let response;
+    try {
+        const { slsEvent, startTime: eventStartTime } = eventInterface.initializeEvent(
+            'kafka',
+            'connect',
+            'produce',
```
Comment (ranrib) [inline src/events/kafkajs.js]: connect

File: src/events/kafkajs.js [hunk 2]
```diff
@@ -93,6 +93,63 @@ function kafkaMiddleware(messages, producer) {
...
+ */
+function kafkaConnectMiddleware(producer) {
+    let originalHandlerAsyncError;
+    let kafkaEvent;
+    let startTime;
+    let response;
+    try {
+        const { slsEvent, startTime: eventStartTime } = eventInterface.initializeEvent(
+            'kafka',
+            'connect',
```
Comment (ranrib) [inline src/events/kafkajs.js]: Should be the hostname

File: src/events/kafkajs.js [hunk 3]
```diff
@@ -93,6 +93,63 @@ function kafkaMiddleware(messages, producer) {
...
+ * @returns {Promise} The response promise
+ */
+function kafkaConnectMiddleware(producer) {
+    let originalHandlerAsyncError;
+    let kafkaEvent;
+    let startTime;
+    let response;
+    try {
+        const { slsEvent, startTime: eventStartTime } = eventInterface.initializeEvent(
+            'kafka',
```
Comment (ranrib) [inline src/events/kafkajs.js]: `kafka_server`, make sure to assign the icon for that in the FE

**Review by ranrib — DISMISSED** [review]

File: src/events/kafkajs.js [hunk 4]
```diff
@@ -93,17 +93,79 @@ function kafkaMiddleware(messages, producer) {
 
     return response;
 }
+/**
+ * Wrap kafka cluster connect function`
```
Comment (galbash) [inline src/events/kafkajs.js]: redundant `

File: src/events/kafkajs.js [hunk 5]
```diff
@@ -93,17 +93,79 @@ function kafkaMiddleware(messages, producer) {
...
+        try {
+            response = originalConnectFunction.apply(this, args);
+        } catch (err) {
+            if (kafkaEvent) {
+                kafkaEvent.setException(kafkaEvent, err);
+                tracer.addEvent(kafkaEvent);
+            }
+            throw err;
+        }
+        response = response.then(res => res).catch((err) => {
```
Comment (galbash) [inline src/events/kafkajs.js]: no need for the `then`

**Review by galbash — APPROVED** [review]

**Pull request merged**